### PR TITLE
Add some utility functions and macros from Infinitesimal Quest 2 + ε

### PR DESCRIPTION
**Pull request by Kodiologist**

### CODE REVIEW

File: hyrule/misc.hy [hunk 1]
```diff
@@ -256,6 +256,26 @@
      (print (.getvalue ~g!hy-s))))
 
 
+(do-mac (do
+  (setv code "
+      (cond
+        (< x 0) -1
+        (> x 0)  1
+        (= x 0)  0
+        True     (raise TypeError))")
```
Comment (scauligi) [inline hyrule/misc.hy:265]: cute

**Review by scauligi — COMMENTED** [review]
Ooh I really like `defmacro-kwargs`, that's very handy.

Although useful, I don't think `sign` really belongs in Hyrule; it's a math utility function that doesn't provide any additional functionality for Hy-specific language items.

**Comment by Kodiologist** [comment]
My intention for Hyrule was never that it be restricted to things that are Hy-specific, exactly; "missing standard functions" are a reasonable use of the library, which is why we have things like `xor`.

**Comment by Kodiologist** [comment]
That's why in the inclusion criteria I only said "It's of general interest to Hy programmers" instead of something like "It's of interest to Hy programmers, but not pure Python programmers".

**Comment by scauligi** [comment]
Hmm, I guess it's still of use if you're writing functional Python, which I can see as another use case for Hyrule.

With that being the case, PR looks good to me 👍

**Pull request merged**